### PR TITLE
Update get_max_neighbor_bins to remove implicit conversion warnings

### PR DESCRIPTION
**Pull request by fredevb**

### CODE REVIEW

File: core/include/traccc/seeding/detail/seeding_config.hpp [hunk 1]
```diff
@@ -114,7 +114,8 @@ struct seedfinder_config {
 
     TRACCC_HOST_DEVICE
     unsigned int get_max_neighbor_bins() const {
-        return std::pow(neighbor_scope[0] + neighbor_scope[1] + 1, 2);
+        return static_cast<unsigned int>(
```
Comment (stephenswat) [inline core/include/traccc/seeding/detail/seeding_config.hpp:117]: If `neighbor_scope` are integers, this is a bad solution. This should be done using integer operations only, e.g.:

```suggestion
        unsigned int t = neighbor_scope[0] + neighbor_scope[1] + 1;
        return t * t;
```

**Review by stephenswat — CHANGES_REQUESTED** [review]

File: core/include/traccc/seeding/detail/seeding_config.hpp (hunk 1 above)
Comment (krasznaa) [reply core/include/traccc/seeding/detail/seeding_config.hpp]: :+1: Absolutely what should be done here.

**Review by krasznaa — CHANGES_REQUESTED** [review]

File: core/include/traccc/seeding/detail/seeding_config.hpp [hunk 2]
```diff
@@ -114,7 +114,9 @@ struct seedfinder_config {
 
     TRACCC_HOST_DEVICE
     unsigned int get_max_neighbor_bins() const {
-        return std::pow(neighbor_scope[0] + neighbor_scope[1] + 1, 2);
+        unsigned int t = neighbor_scope[0] + neighbor_scope[1] + 1;
+        return t * t;
+            std::pow(neighbor_scope[0] + neighbor_scope[1] + 1, 2));
```
Comment (krasznaa) [inline core/include/traccc/seeding/detail/seeding_config.hpp:119]: :frowning: This line was left in by mistake. (Stephen didn't set up his comment to be applied as-is.)

**Review by stephenswat — APPROVED** [review]
Looks good like this.

**Comment by stephenswat** [comment]
On a side node, please keep commit message titles under 72 characters.

**Pull request merged**